UIKitTemplate/UIKitTemplate/Controllers/FeedViewController.swift

```diff
@@ -0,0 +1,124 @@
+        super.viewDidLoad()
+        setupNavigationItem()
+        setupView()
+    }
+
+    // MARK: - Private Methods
+
+    private func setupView() {
+        feedTableView.addSubview(refreshControl)
+        view.addSubview(feedTableView)
+
+        setupContraints()
+    }
+
+    private func setupContraints() {
+        feedTableView.translatesAutoresizingMaskIntoConstraints = false
+        [
+            feedTableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            feedTableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            feedTableView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            feedTableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ].activate()
+    }
+
+    private func setupNavigationItem() {
+        let messagesBarItem = UIBarButtonItem(image: .messagesBarIcon.withRenderingMode(.alwaysOriginal))
+        let logoBarItem = UIBarButtonItem(image: .logo.withRenderingMode(.alwaysOriginal))
+        navigationItem.setRightBarButton(messagesBarItem, animated: false)
+        navigationItem.setLeftBarButton(logoBarItem, animated: false)
+    }
+
+    @objc private func handleRefresh(_ refreshControl: UIRefreshControl) {
+        refreshControl.endRefreshing()
+    }
+}
+
+// MARK: - UITableViewDelegate
+
+extension FeedViewController: UITableViewDelegate {}
+
+// MARK: - UITableViewDataSource
+
+extension FeedViewController: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch sections[section] {
+        case .stories, .recommendation:
+            return 1
+        case let .post(isFirst) where isFirst:
+            return 1
+        case .post:
+            return AppDataProvider.shared.posts.count - 1
+        }
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch sections[indexPath.section] {
+        case .stories:
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: StoriesCell.reuseID) as? StoriesCell
+            else { return UITableViewCell() }
+            cell.stories = AppDataProvider.shared.stories
+            return cell
+        case let .post(isFirst):
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: PostCell.reuseID) as? PostCell
+            else { return UITableViewCell() }
+            let post = AppDataProvider.shared.getPost(byIndex: isFirst ? indexPath.row : indexPath.row + 1)
+            cell.setupCell(withPost: post)
+            return cell
+        case .recommendation:
+            guard let cell = tableView
+                .dequeueReusableCell(withIdentifier: RecommendationCell.reuseID) as? RecommendationCell
+            else {
+                return UITableViewCell()
+            }
```

Comment:
```suggestion
            guard let cell = tableView
                .dequeueReusableCell(withIdentifier: RecommendationCell.reuseID) as? RecommendationCell else { return UITableViewCell() }
```
такие вещи на одно строчки пишем если умещаются